frontend/packages/client/src/hooks/useVotesForAddress.js

```diff
@@ -1,41 +1,40 @@
-import { useCallback, useReducer } from 'react';
+import { useErrorHandlerContext } from 'contexts/ErrorHandler';
 import { checkResponse } from 'utils';
-import { INITIAL_STATE, defaultReducer } from '../reducers';
+import { useQuery } from '@tanstack/react-query';
 
-export default function useVotesForAddress() {
-  const [state, dispatch] = useReducer(defaultReducer, {
-    ...INITIAL_STATE,
-    loading: false,
-  });
+export default function useVotesForAddress({
+  proposalIds,
+  addr,
+  enabled = true,
+} = {}) {
+  const { notifyError } = useErrorHandlerContext();
+  const { isLoading, isError, data, error } = useQuery(
+    ['user-votes', addr, proposalIds],
+    async () => {
+      const response = await fetch(
```

Comment:
Same here - lets move this one to a function in our `api` directory

Reply:
updated